librosa/feature/spectral.py

```diff
@@ -849,19 +855,26 @@ def rms(y=None, S=None, frame_length=2048, hop_length=512,
                        frame_length=frame_length,
                        hop_length=hop_length)
 
-        # No normalization is necessary for time-domain input
-        norm = 1
+        # Calculate power
+        power = np.mean(np.abs(x)**2, axis=0, keepdims=True)
     elif S is not None:
         x, _ = _spectrogram(y=y, S=S,
                             n_fft=frame_length,
                             hop_length=hop_length)
+        # power spectrogram
+        x = np.abs(x) ** 2
+
+        # Recover negative frequency bins' power
+        end = None if odd_n_fft else -1
+        x[1:end] *= 2
 
-        # FFT introduces a scaling of n_fft to energy calculations
-        norm = 2 * (x.shape[0] - 1)
+        # Calculate power
+        n_fft = x.shape[0]*2 - 1 if odd_n_fft else x.shape[0]*2 - 2
+        power = np.sum(x, axis=0, keepdims=True, dtype=np.float64) / n_fft**2
```

Comment:
power should not explicitly set the dtype here; it should inherit from the dtype of the input.

Reply:
Thank you for pointing it out. I fixed it not to specify dtype.